Composer/packages/client/src/components/Split/LeftRightSplit.tsx

```diff
@@ -0,0 +1,232 @@
+  }
+
+  // ensure within total bounds after adjustments
+  return Math.max(0, Math.min(newPrimary, total - splitter));
+};
+
+type Props = {
+  initialLeftGridWidth?: string | number;
+  minLeftPixels?: number;
+  minRightPixels?: number;
+  splitterWidth?: number;
+  renderSplitter?: () => JSX.Element;
+};
+
+export const LeftRightSplit = (props: React.PropsWithChildren<Props>): JSX.Element => {
+  const {
+    initialLeftGridWidth: defaultLeftWidth,
+    minRightPixels,
+    minLeftPixels,
+    splitterWidth = defaultSplitterWidth,
+    renderSplitter,
+  } = props;
+
+  const [currentContentWidth, setCurrentContentWidth] = React.useState<number>(0);
+  const [currentLeftWidth, setCurrentLeftWidth] = React.useState<number>(0);
+
+  const [leftWidth, setLeftWidth] = React.useState(() => {
+    // If the default is a number, then use it or the left minimum as a value.
+    const numericValue = Number(defaultLeftWidth);
+    return isNaN(numericValue) ? -1 : Math.max(numericValue, minLeftPixels ?? numericValue);
+  });
+
+  const [leftStart, setLeftStart] = React.useState(0);
+  const [screenStart, setScreenStart] = React.useState(0);
+
+  const constrainLeft = (value: number): number => {
+    return constrainPaneExtent(value, {
+      total: currentContentWidth,
+      splitter: splitterWidth,
+      minPrimary: minLeftPixels,
+      minSecondary: minRightPixels,
+    });
+  };
+
+  React.useEffect(() => {
+    if (leftWidth !== -1) {
+      const newLeft = constrainLeft(leftWidth);
+      setLeftWidth(newLeft);
+    }
+  }, [currentContentWidth, splitterWidth]);
+
+  const onContentMeasure = (contentRect: ContentRect) => {
+    contentRect.bounds && setCurrentContentWidth(contentRect.bounds.width);
+  };
+
+  const onLeftMeasure = (contentRect: ContentRect) => {
+    contentRect.bounds && setCurrentLeftWidth(contentRect.bounds.width);
+  };
+
+  const onSplitMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+    event.currentTarget.setPointerCapture(1);
+    setScreenStart(event.screenX);
+    setLeftStart(currentLeftWidth);
+  };
+
+  const onSplitMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (event.currentTarget.hasPointerCapture(1)) {
+      // calculate candidate left
+      const newLeft = constrainLeft(leftStart + (event.screenX - screenStart));
+      setLeftWidth(newLeft);
+    }
+  };
+
+  const onSplitMouseUp = (event: React.MouseEvent<HTMLDivElement>) => {
+    event.currentTarget.releasePointerCapture(1);
+  };
+
+  // If the left width has never been set
+  // try using the default as a number and constraining it
+  let renderLeftWidth = `${leftWidth}px`;
+  if (leftWidth < 0) {
+    renderLeftWidth = toGridWidth(defaultLeftWidth);
+  }
+
+  const children = React.Children.toArray(props.children);
+  const leftChild = children.length > 0 ? children[0] : <div />;
+  const rightChild = children.length > 1 ? children[1] : <div />;
+
+  const renderSplitVisual =
+    renderSplitter ??
```

Comment:
When custom rendering the splitter, isn't the width of it important in your calculations? (L#27)

Reply:
No - it should render within whatever area it is given by the splitterWidth just like panes do. The caller provided the splitterWidth param so they should have it for the custom render as well.